### PR TITLE
CC-2076 Set permissions and group for binaries and default config

### PR DESCRIPTION
**Pull request by glujan**
[Jira](https://jira.zenoss.com/browse/CC-2076)

### CODE REVIEW

**Comment by jpeacock-zenoss** [comment]
I can't tell for certain, but does this change ensure that everything in `/opt/serviced` has `root:serviced` ownership?

File: pkg/deb/postinstall [hunk 1]
```diff
@@ -2,3 +2,9 @@
 mkdir -p /var/log/serviced
 chmod 1777 /var/log/serviced
 
+for BINARY in "serviced" "serviced-controller" "serviced-storage"; do
+    chgrp serviced /opt/serviced/bin/${BINARY}
+done
+
+chgrp serviced /etc/default/serviced
+chgrp serviced /opt/serviced
```
Comment (jpeacock-zenoss) [inline pkg/deb/postinstall]: There's a loop above that changes three specific files to have `root:serviced` ownership, but it seems inconsistent to have different ownership in the `/opt/serviced` directory.

Changing the above line to `chgrp -R serviced /opt/serviced` will switch everything over to the `serviced` group and eliminate the need for the earlier loop.

Comment (glujan) [inline pkg/deb/postinstall]: You are right, it makes sense to recursively change group. And permissions are already set in `makefile`.

**Comment by glujan** [comment]
@tgjonestx Could you have a look at this as well? Does this PR make sense to you?

**Comment by tgjonestx** [comment]
@glujan - A few comments

* I don't see where the permissions for `/etc/default/serviced` are set to `640`
  * We don't want users who are not in the `serviced` group to view the configuration
* I don't see where the permissions for `/opt/serviced` are set to `750`
  * We don't want users who are not in the `serviced` group to list any directory contents below `/opt/serviced`
  * We don't want user who are not in the `serviced` group to be able to execute the binaries in `/opt/serviced/bin` 
* I don't see where the `root` user is added to the `serviced` group
  * Technically speaking this is not required to achieve our security goals, but I think it's a nice-to-have to make that `serviced` group somewhat "self documenting"; i.e. these are the users allowed to use serviced and by default, only root is allowed

Also, did you trying building and installing the RPM to verify that the group and permissions are all set as expected?

File: makefile [hunk 2]
```diff
@@ -410,7 +413,8 @@ $(install_DIRS): FORCE
 				exit 1 ;\
 			fi ;\
 		done ;\
-	done
+	done ;\
+	chown 750 $(_DESTDIR)${prefix}
```
Comment (glujan) [inline makefile]: Here permissions are set to `/opt/serviced/` directory

**Comment by glujan** [comment]
Retest this

**Comment by glujan** [comment]
retest

**Pull request merged**